### ISSUE TITLE
Unbound fails to start with multiple ipv6 interfaces 

### ISSUE THREAD

**Issue by rush2049** [NLnetLabs/unbound#545]
**Describe the bug**
Inside pfsense 2.5.2-RELEASE unbound will fail to start if the following is true:

- ipv6 enabled
- LAN interface IPv6 Configuration type set to track interface (WAN)
- LAN interface IPv6 Prefix ID set to 0
- Guest interface (a vlan under LAN) IPv6 Configuration type set to track interface (WAN)
- Guest interface IPv6 Prefix ID set to 1

If the Pfsense box is then rebooted in order for the Guest interface to pickup an IPv6 address/scope then unbound will no longer start.

The workaround is to deselect the guest interface completely in the  interface selection of unbound. But then you no longer have DNS services on the guest network (the vlan interface).
Or another solution is to completely remove IPv6 from the guest vlan interface. Thus having IPv4 and IPv6 resolution on the main LAN interface, but only IPv4 resolution on the guest vlan interface.

The relevant section of the unbound config looks like this without the guest vlan interface bind:
```
# Interface IP(s) to bind to
interface: 192.168.10.68
interface: 192.168.10.68@853
interface: 2601:981:c201:ba00:a236:9fff:fe8a:298a
interface: 2601:981:c201:ba00:a236:9fff:fe8a:298a@853
interface: 127.0.0.1
interface: 127.0.0.1@853
interface: ::1
interface: ::1@853
```

The same section with the guest vlan interface bind enabled:
```
# Interface IP(s) to bind to
interface: 192.168.10.68
interface: 192.168.10.68@853
interface: 2601:981:c201:ba00:a236:9fff:fe8a:298a
interface: 2601:981:c201:ba00:a236:9fff:fe8a:298a@853
interface: 192.168.11.68
interface: 192.168.11.68@853
interface: 2601:981:c201:ba01:a236:9fff:fe8a:298a
interface: 2601:981:c201:ba01:a236:9fff:fe8a:298a@853
interface: 127.0.0.1
interface: 127.0.0.1@853
interface: ::1
interface: ::1@853
```

The error message logged by unbound:

`The command '/usr/local/sbin/unbound -c /var/unbound/unbound.conf' returned exit code '1', the output was '[1632803036] unbound[75182:0] error: can't bind socket: Can't assign requested address for 2601:981:c201:ba01:a236:9fff:fe8a:298a port 53 [1632803036] unbound[75182:0] fatal error: could not open ports'`

Followed by a lot of these messages:

`Could not deliver signal HUP to process because its pidfile (/var/run/unbound.pid) cannot be read, No such file or directory.`

**System:**
 - Unbound version: whatever is included alongside pfsense 2.5.2-RELEASE

**Additional information**
Add any other information that you may have gathered about the issue here.

**Comment by wcawijngaards** [comment]
The issue is exactly what the error reports, the IPv6 address 2601... can not be opened. Because it is present twice in the list of interfaces, the second time it cannot be opened because it is already open. Remove the second interface option that is the identical IPv6 address.
So, delete these two lines (the second time they appear) from the config
```
interface: 2601:981:c201:ba01:a236:9fff:fe8a:298a
interface: 2601:981:c201:ba01:a236:9fff:fe8a:298a@853
```

In recent versions of unbound, a new feature has been introduced where it is possible to bind to interfaces by naming them, instead of listing the IP addresses. Not sure if that helps, it depends on the IP addresses that are part of the interface names; if you changes to use names and that contained the same IPv6 address twice you would have the same error.

**Comment by rush2049** [comment]
If you look more closely you'll see that the ipv6 addresses are not the same. They used different IPv6 prefixes and are using a different number in the fourth grouping of the address.

2601:981:c201:       ba00    :a236:9fff:fe8a:298a

2601:981:c201:       ba01    :a236:9fff:fe8a:298a

**Comment by wcawijngaards** [comment]
You are right, so that is not the problem. Something else must have the IP address `2601:981:c201:ba01:a236:9fff:fe8a:298a` open and that is why unbound cannot open the interface. Some other server I guess, on the machine.

**Comment by rush2049** [comment]
Is it a conflict with the way pfsense handles vlan interfaces?

I was more under the impression that unbound did not adequately check to see if the ipv6 was unique quite like you did. Possibly a bug, this why I posted about it here.

**Comment by dyunwei** [comment]
You can check if the IPv6 address has been opened with "netstat  -6 -an" command.  I guess it should be opened by some other process.
